main.go

```diff
@@ -49,14 +49,23 @@ func main() {
 	}
 
 	// Let's convert the list of Todos to JSON
-	todoJsonBytes, err := json.MarshalIndent(todos, "", "    ")
+	combinedTodos := append(existingTodos, todos...)
+	todoJsonBytes, err := json.MarshalIndent(combinedTodos, "", "    ")
 	if err != nil {
 		fmt.Printf("Error when marshalling to JSON: %v", err)
 		return
 	}
 
 	todoJsonContents := string(todoJsonBytes)
 
+	//Rewriting the whole todo list into a single JSON object instead of creating new object everytime
+	//Might use a different approach with seek, but this is simpler(?)s
+	err = f.Truncate(0)
+
```

Comment:
You also need to do a fseek(0, 0) or it won't work.

Reply:
Since we are passing 0 to Truncate, it should work. I tested again and it did not fail in any case. Could you tell in which case it would fail without fseek?